component/encryptor/encryptor.go

```diff
@@ -0,0 +1,382 @@
+	return objAttrs, "", nil
+}
+
+func checkForActualFileSize(fileHandle *os.File, currentFileSize int64, blockSize int64) (int64, error) {
+
+	totalBlocks := currentFileSize / (blockSize + MetaSize)
+	if currentFileSize < totalBlocks*(blockSize+MetaSize)+8 {
+		return 0, nil
+	}
+
+	// Read the last 8 bytes of file and check for padding
+	paddingLengthBytes := make([]byte, 8)
+	_, err := fileHandle.ReadAt(paddingLengthBytes, currentFileSize-8)
+	if err != nil {
+		log.Err("Encryptor: Error reading last 8 bytes of file: %s", err.Error())
+		return 0, err
+	}
+
+	actualFileSize := currentFileSize - int64(binary.BigEndian.Uint64(paddingLengthBytes)) - MetaSize - 8
+	return actualFileSize, nil
+}
+
+func (e *Encryptor) GetAttr(options internal.GetAttrOptions) (attr *internal.ObjAttr, err error) {
+	// Open the file from the mount point and get the attributes.
+	log.Info("Encryptor::GetAttr for %s", options.Name)
+	fileAttr, err := os.Stat(e.mountPointCipher + options.Name)
+	if err != nil {
+		if os.IsNotExist(err) {
+			log.Trace("Encryptor::GetAttr : File not found: %s", options.Name)
+			return nil, syscall.ENOENT
+		}
+		log.Trace("Encryptor::GetAttr : Error getting file attributes: %s", err.Error())
+		return &internal.ObjAttr{}, nil
+	}
+
+	// Populate the ObjAttr struct with the file info
+	attr = &internal.ObjAttr{
+		Mtime:  fileAttr.ModTime(),                // Modified time
+		Atime:  time.Now(),                        // Access time (current time as approximation)
+		Ctime:  fileAttr.ModTime(),                // Change time (same as modified time in this case)
+		Crtime: fileAttr.ModTime(),                // Creation time (not available in Go, using modified time)
+		Mode:   fileAttr.Mode(),                   // Permissions
+		Path:   e.mountPointCipher + options.Name, // Full path
+		Name:   fileAttr.Name(),                   // Base name of the path
+	}
+	fileHandle, err := os.OpenFile(e.mountPointCipher+options.Name, os.O_RDONLY, 0666)
+	if err != nil {
+		log.Trace("Encryptor::GetAttr : Error opening file: %s", err.Error())
+		return nil, err
+	}
+	defer fileHandle.Close()
+
+	if fileAttr.IsDir() {
+		attr.Size = fileAttr.Size()
+		attr.Flags.Set(internal.PropFlagIsDir)
+	} else {
+		actualFileSize, err := checkForActualFileSize(fileHandle, fileAttr.Size(), int64(e.blockSize))
+		if err != nil {
+			log.Err("Encryptor: Error checking for actual file size: %s", err.Error())
+			return nil, err
+		}
+		log.Debug("Encryptor::GetAttr : actual file size %d", actualFileSize)
+		attr.Size = actualFileSize
+	}
+	return attr, nil
+}
+
+func (e *Encryptor) StageData(opt internal.StageDataOptions) error {
+	log.Info("Encryptor::StageData : %s, offset %d, data %d", opt.Name, opt.Offset, len(opt.Data))
+	return encryptWriteBlock(e.handle, opt.Name, opt.Data, opt.Offset, e.blockSize, e.encryptionKey)
+}
+
+func encryptWriteBlock(encryptedFile *os.File, name string, data []byte, blockId uint64, blockSize uint64, encryptionKey []byte) error {
```

Comment:
any particular reason this is a func and not logic directly within StageData?

Reply:
yes wanted to indicate stage is internally encrypting. may be i can add comment for this and move this to stage method.